lib/cocoapods/installer/analyzer/analysis_result.rb

```diff
@@ -51,6 +53,29 @@ def all_user_build_configurations
             result.merge(target.user_build_configurations)
           end
         end
+
+        # @return [Bool] Whether an installation should be performed or this
+        #         CocoaPods project is already up to date.
+        #
+        def needs_install?
+          podfile_needs_install? || sandbox_needs_install?
+        end
+
+        # @return [Bool] Whether the podfile has changes respect to the lockfile.
+        #
+        def podfile_needs_install?
+          state = podfile_state
+          needing_install = state.added + state.changed + state.deleted
+          !needing_install.empty?
+        end
+
+        # @return [Bool] Whether the sandbox is in synch with the lockfile.
+        #
+        def sandbox_needs_install?
+          state = sandbox_state
+          needing_install = state.added + state.changed + state.deleted
+          !needing_install.empty?
```

Comment:
nit: Seems like these could just use the count rather than creating a new array

```
state = sandbox_state
needing_install = state.added.length + state.changed.length + state.deleted.length
needing_install > 0
```

Reply:
done